src/internal/document_api/requests.rs

```diff
@@ -456,32 +474,38 @@ pub mod document_access {
                     from_public_key: from_pub_key.clone().into(),
                     to: req_grants,
                 };
-                Box::new(auth.request.post(
-                    &format!("documents/{}/access", rest::url_encode(id.id())),
-                    &req,
-                    RequestErrorCode::DocumentGrantAccess,
-                    AuthV2Builder::new(&auth, Utc::now()),
-                ))
+                auth.request
+                    .post(
+                        &format!("documents/{}/access", rest::url_encode(id.id())),
+                        &req,
+                        RequestErrorCode::DocumentGrantAccess,
+                        AuthV2Builder::new(&auth, Utc::now()),
+                    )
+                    .compat()
+                    .await
             }
             // the failure case here is that we couldn't convert the recrypt EncryptedValue because
             // it was not an EncryptedOnceValue -- really just a limitation of Rust's enums as we expect these to be EncryptedOnceValues
-            Err(e) => Box::new(futures::future::failed(e)),
+            Err(e) => Err(e),
```

Comment:
You can also get rid of this one with a `?` on the end of 469.

Reply:
Similar issue to the previous collect, but still much improved!